pkg/authorization/apibinding_authorizer.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authorization
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kcp-dev/logicalcluster"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	clientgoinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	klog "k8s.io/klog/v2"
+	"k8s.io/kubernetes/plugin/pkg/auth/authorizer/rbac"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	kcpinformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	rbacwrapper "github.com/kcp-dev/kcp/pkg/virtual/framework/wrappers/rbac"
+)
+
+const (
+	byWorkspaceIndex = "apiBindingAuthorizer-byWorkspace"
+)
+
+// NewAPIBindingAccessAuthorizer returns an authorizer that checks for
+func NewAPIBindingAccessAuthorizer(versionedInformers clientgoinformers.SharedInformerFactory, f kcpinformers.SharedInformerFactory, delegate authorizer.Authorizer) authorizer.Authorizer {
+	if _, found := f.Apis().V1alpha1().APIBindings().Informer().GetIndexer().GetIndexers()[byWorkspaceIndex]; !found {
+		err := f.Apis().V1alpha1().APIBindings().Informer().AddIndexers(
+			cache.Indexers{byWorkspaceIndex: func(obj interface{}) ([]string, error) {
+				return []string{logicalcluster.From(obj.(metav1.Object)).String()}, nil
+			},
+			},
+		)
+		if err != nil {
+			// nothing we can do here. But this should also never happen. We check for existence before.
+			klog.Errorf("failed to add indexer for APIBindings: %v", err)
+		}
+	}
+
```

Comment:
this is missing to call `Rbac().V1().{all the resources we need}`. Background: informers only start those watches for the resource where at least one `Rbac().V1().Resource().Informers()` was called.

Reply:
done